packages/babel-helpers/src/helpers/usingCtx.js

```diff
@@ -0,0 +1,78 @@
+/* @minVersion 7.23.0 */
+
+export default function _usingCtxFactory() {
+  var dispose_SuppressedError;
+  if (typeof SuppressedError !== "undefined") {
+    // eslint-disable-next-line no-undef
+    dispose_SuppressedError = SuppressedError;
+  } else {
+    dispose_SuppressedError = function SuppressedError(suppressed, error) {
+      this.suppressed = suppressed;
+      this.error = error;
+      this.stack = new Error().stack;
+    };
+    dispose_SuppressedError.prototype = Object.create(Error.prototype, {
+      constructor: {
+        value: dispose_SuppressedError,
+        writable: true,
+        configurable: true,
+      },
+    });
+  }
+
+  return (_usingCtxFactory = function _usingCtx() {
+    var empty = {},
+      stack = [];
+    return {
+      // error
+      e: empty,
+      // using
+      u: function (value, isAwait) {
+        if (value === null || value === void 0) return value;
+        if (typeof value !== "object") {
+          throw new TypeError(
+            "using declarations can only be used with objects, null, or undefined."
+          );
+        }
+        // core-js-pure uses Symbol.for for polyfilling well-known symbols
+        if (isAwait) {
+          var dispose =
+            value[Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")];
+        }
+        if (dispose === null || dispose === void 0) {
+          dispose = value[Symbol.dispose || Symbol.for("Symbol.dispose")];
+        }
+        if (typeof dispose !== "function") {
+          throw new TypeError(`Property [Symbol.dispose] is not a function.`);
+        }
+        stack.push({ v: value, d: dispose, a: isAwait });
+        return value;
+      },
+      // dispose
+      d: function () {
+        var error = this.e;
+
+        function next() {
+          while ((r = stack.pop())) {
+            try {
+              var r,
+                p = r.d.call(r.v);
```

Comment:
The `var r` here is incredibly confusing, I was initially surprised that it's not `undefined` :P

Unfortunately Terser cannot do this optimization for us so it's fine to do it manually, but I opened https://github.com/terser/terser/issues/1446.

Also, let's rename `r` to `resource` and `p` to `disposalResult` since we now mangle variables anyway 😉